### DISCUSSION TITLE
Formstate "touched" is not working while using React-Select

### DISCUSSION THREAD

**Discussion by anjana543** [react-hook-form/react-hook-form#4196] [Q&A]
Hi @bluebill1049,

There is an issue while getting the "touched" object of form state in react hook form. 

Issue Description:
-------------------------
**Added 3 fields** - reactSelect (using React-select), controllerSelect(using controller), lastname (normal input)
**Output** - touched object has only one input field "lastname" after entering all the 3 fields
**Expected Result** - Show all 3 fields in the touched object when values are added/updated
![image](https://user-images.githubusercontent.com/4394526/107887197-5a58a300-6f2a-11eb-837c-990876f2d961.png)

Please find the below link to reproduce the issue,
https://codesandbox.io/s/amazing-http-77rx6?file=/src/App.js

Thank you.

**Comment by bluebill1049** [comment]
This could only happy when onBlur is not triggered and the hook form is not aware of the onBlur event. you may want to use `useController`?

**Reply by anjana543** [reply]
@bluebill1049  How to use the OnChange of use controller? Could you please explain using the below codesandbox example?
https://codesandbox.io/s/amazing-http-77rx6?file=/src/App.js

I tried using controller "as" and it worked. However, onChange event is not working with V6. Is there any option to add onChange for controller "as"??

**Reply by bluebill1049** [reply]
you can do `render`

```
render={({ onChange, onBlur, value }) => {
  // you just have to write them up
}}
```

have a look here: https://react-hook-form.com/api#Controller

**Reply by anjana543** [reply]
@bluebill1049  I tried this but "touched" state is not working. Could you please help me in understanding the use Controller with the below example,
https://codesandbox.io/s/amazing-http-77rx6?file=/src/App.js

In the above sample code, you can see that I've used controller render onChange event but the field is not getting added in the touched object. Even I tried adding setValue to the change event but still, it's not helping.
Please advise!

Thanks.

**Reply by bluebill1049** [reply]
<img width="1262" alt="Screen Shot 2021-02-15 at 4 15 47 pm" src="https://user-images.githubusercontent.com/10513364/107907966-1658b300-6fa9-11eb-8262-c6f973801b78.png">

you only missed `onBlur` :)

**Reply by bluebill1049** [reply]
The first example is not going to work because it's a customer register.

useEffect(() => {
    register({ name: "reactSelect" });
  }, [register]);


until V7 (next major) which it's possible.

**Reply by bluebill1049** [reply]
v7

```
const { onChange, onBlur, ref, name } =  register('test');
```

you can use the onBlur to update the touched

**Reply by anjana543** [reply]
> <img alt="Screen Shot 2021-02-15 at 4 15 47 pm" width="1262" src="https://user-images.githubusercontent.com/10513364/107907966-1658b300-6fa9-11eb-8262-c6f973801b78.png">
> 
> you only missed `onBlur` :)

Thank you so much @bluebill1049!



> The first example is not going to work because it's a customer register.
> 
> useEffect(() => {
> register({ name: "reactSelect" });
> }, [register]);
> 
> until V7 (next major) which it's possible.

When are you going to release v7?

**Reply by bluebill1049** [reply]
It's in beta right now, i hope in the next 2-3 weeks we can go into RC version.

**Reply by bluebill1049** [reply]
here is the RFC if you interested: https://github.com/react-hook-form/react-hook-form/discussions/3714
